### DISCUSSION TITLE
Add ability to change shell executor for language servers/internal processes

### DISCUSSION THREAD

**Discussion by Tefached95** [helix-editor/helix#10233] [Troubleshooting]
Trying to set up `erlang-ls` on Windows, I compiled it and generated the required binary, which I placed in my `PATH`. Running it in bash works properly. However, trying to execute the binary from the command prompt or PowerShell results in a "not a proper executable" error.

Curiously, the `which` rust crate will not find an extensionless executable in the `PATH` despite the fact that running `which erlang-ls` properly finds the file.

This can be "fixed" by adding an `.exe` extension manually but when opening an erlang file, the language server fails to start, and the log shows the following line:

```log
2022-12-09T15:44:13.904 helix_view::editor [ERROR] Failed to initialize the LSP for `source.erlang` { IO Error: This version of %1 is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher. (os error 216) }
```

My proposal is to make the shell executor configurable for internal processes, such as running the language servers, something akin to the `editor.shell` property. Maybe I'm going about this the wrong way but so far I haven't been able to find a workaround for running `erlang-ls` on Windows in helix.

**Comment by archseer** [comment]
There is no shell used, the server is started directly as a subprocess.

**Comment by Tefached95** [comment]
Should we then consider this a bug? At least in my eyes this is a valid use case and it's not working as expected.

**Comment by the-mikedavis** [comment]
Is Helix built for Windows on your machine? I'm not very familiar with Windows or WSL but we may not be able to call a WSL program from a Windows program in the same way that command prompt / PowerShell can't execute erlang-ls.

FWIW the code to execute the language server as a child process is here: https://github.com/helix-editor/helix/blob/bdeefbfb23077fcbbfe1e7df6c6ac88516244bbc/helix-lsp/src/client.rs#L59-L67
